internal/namespace/caveats.go

```diff
@@ -14,7 +14,16 @@ import (
 // definition, including usage of the parameters.
 func ValidateCaveatDefinition(caveat *core.CaveatDefinition) error {
 	// Ensure all parameters are used by the caveat expression itself.
-	deserialized, err := caveats.DeserializeCaveat(caveat.SerializedExpression)
+	parameterTypes, err := caveattypes.DecodeParameterTypes(caveat.ParameterTypes)
```

Comment:
do you foresee this could cause existing deployments to throw this error somehow now that we add the extra layer of type validation?

Reply:
Yes, this is possible. I plan to put a warning in the release notes

Reply:
but how would existing deployments fix it?

Reply:
they'd rewrite their caveat expression

Reply:
how would they tell ahead of time their caveat is "incorrect"

Reply:
You can't; same way you can't tell today if it is incorrect if you take in a dynamic value and it type check fails at runtime. It would be the exact same thing: you passed a wrong value in context and it failed to eval, causing a type error